server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2728,6 +2729,9 @@ private IndexWriterConfig getIndexWriterConfig() {
         }
         if (config().getIndexSort() != null) {
             iwc.setIndexSort(config().getIndexSort());
+            if (engineConfig.hasNested()) {
+                iwc.setParentField(ROOT_DOC_FIELD_NAME);
```

Comment:
I wonder how this works when the nested field is added as an update to the mapping? Should we always set the parent field in case a nested field is added later?

Reply:
Good point, I was thinking about this but wasn't sure of the implications. @jpountz any thoughts on this?

Reply:
I asked a similar question to @s1monw a while back, and he suggested always adding the parent field when an index sort is configured.

Reply:
Excellent, this simplifies things. Lemme update the PR.

Reply:
Done.

Reply:
Maybe include the reason why parent field is always enabled in the comment too?

Reply:
Right, this was added in the last iteration.